nixos/modules/virtualisation/parallels-guest.nix

```diff
@@ -45,34 +47,36 @@ in
 
   config = mkIf config.hardware.parallels.enable {
     services.xserver = {
-      drivers = singleton
-        { name = "prlvideo"; modules = [ prl-tools ]; };
+      videoDrivers = [ "prlvideo" ];
 
-      screenSection = ''
-        Option "NoMTRR"
-      '';
+      modules = [ prl-tools ];
 
       config = ''
         Section "InputClass"
-          Identifier "prlmouse"
-          MatchIsPointer "on"
-          MatchTag "prlmouse"
-          Driver "prlmouse"
+          Identifier      "prlmouse"
+          MatchIsPointer  "on"
+          MatchTag        "prlmouse"
+          Driver          "prlmouse"
         EndSection
       '';
+
+      screenSection = ''
+        Option "NoMTRR"
+      '';
     };
 
     hardware.opengl.package = prl-tools;
     hardware.opengl.package32 = pkgs.pkgsi686Linux.linuxPackages.prl-tools.override { libsOnly = true; kernel = null; };
-    hardware.opengl.setLdLibraryPath = true;
+    hardware.opengl.extraPackages = [ pkgs.mesa.drivers ];
```

Comment:
Why are they needed when we use Parallels driver?

Reply:
Same question about OpenGL here -- do we need to set `opengl.package` at all now?

Reply:
Seems that we should use the [VirGL 3D driver](https://virgil3d.github.io/) provided by Mesa instead.

Reply:
Great! @wegank @paulsmith can you try commenting out all `hardware.opengl` stuff and verifying that OpenGL still works for you?

Reply:
@abbradar Yes, it works.

```
[nix-shell:~]$ glxinfo | grep OpenGL
OpenGL vendor string: Mesa/X.org
OpenGL renderer string: virgl
OpenGL core profile version string: 4.0 (Core Profile) Mesa 21.3.2
OpenGL core profile shading language version string: 4.00
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.1 Mesa 21.3.2
OpenGL shading language version string: 1.40
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.0 Mesa 21.3.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
OpenGL ES profile extensions:

[nix-shell:~]$ glxinfo | grep rendering
direct rendering: Yes
```